### PR TITLE
Update creature_template and creature_template_scaling

### PR DESCRIPTION
**Pull request by funjoker**
**Changes proposed:**

-  Changed family from uint8 to uint32 (packet says uint32)
-  Update ContentTuning DB
-  Update Class Level stats 

**Target branch(es):** 3.3.5/master

- [x] bfa

### CODE REVIEW

**Comment by Aokromes** [comment]
add sql to pr!

**Comment by funjoker** [comment]
Damage sent in CombatLog musst be the upscaled damage.
Client calculates the rest itself

File: src/server/game/Globals/ObjectMgr.h [hunk 1]
```diff
@@ -1290,6 +1290,7 @@ class TC_GAME_API ObjectMgr
 
         CreatureBaseStats const* GetCreatureBaseStats(uint8 level, uint8 unitClass);
 
+
```
Comment (GitVerge) [inline src/server/game/Globals/ObjectMgr.h]: hm?

Comment (illfated) [reply src/server/game/Globals/ObjectMgr.h]: I highly doubt it. Likely a glitch.

**Comment by funjoker** [comment]
Will clean Things up

**Comment by Aokromes** [comment]
@funjoker this is completed? if yes merge the 2 sql plz.

**Comment by funjoker** [comment]
that's very experimental but far better than how it is actually in tc.
Some calculations according to damage etc is weird

**Comment by funjoker** [comment]
Also need to add fallback methodic since creatures in dungeons use highest difficulty <= map difficulty.
If HC dungeon and creatre only has normal difficulty scaling it should use normal.
If Mythic and creature has max hc difficulty scaling it should use hc and not normal.
etc.

**Comment by Aokromes** [comment]
Conflicting files
src/server/game/Entities/Creature/Creature.cpp
src/server/game/Globals/ObjectMgr.cpp
src/server/game/Server/Packets/CombatLogPacketsCommon.cpp

btw, it's still needed after 8.1.5 changes?

**Comment by funjoker** [comment]
yes

File: src/server/game/Globals/ObjectMgr.cpp [hunk 2]
```diff
@@ -3636,42 +3641,70 @@ void ObjectMgr::LoadPlayerInfo()
...
-            uint32 current_level = fields[2].GetUInt8();
+            uint32 current_level = fields[1].GetUInt8();
             if (current_level > sWorld->getIntConfig(CONFIG_MAX_PLAYER_LEVEL))
             {
                 if (current_level > STRONG_MAX_LEVEL)        // hardcoded level maximum
-                    TC_LOG_ERROR("sql.sql", "Wrong (> %u) level %u in `player_levelstats` table, ignoring.", STRONG_MAX_LEVEL, current_level);
+                    TC_LOG_ERROR("sql.sql", "Wrong (> %u) level %u in `player_classlevelstats` table, ignoring.", STRONG_MAX_LEVEL, current_level);
                 else
                 {
                     TC_LOG_INFO("misc", "Unused (> MaxPlayerLevel in worldserver.conf) level %u in `player_levelstats` table, ignoring.", current_level);
```
Comment (ReyDonovan) [inline src/server/game/Globals/ObjectMgr.cpp]: `player_classlevelstats`

**Comment by Aokromes** [comment]
Conflicting files
src/server/game/DataStores/GameTables.h
src/server/game/Entities/Creature/Creature.cpp
src/server/game/Globals/ObjectMgr.cpp
src/server/game/Server/Packets/CombatLogPacketsCommon.cpp

**Comment by illfated** [comment]
Maybe the coming weekend will be a better time for @funjoker to do some follow-up?

**Pull request closed without merging**